src/wtforms/validators.py

```diff
@@ -322,14 +322,16 @@ class Regexp(object):
         Error message to raise in case of a validation error.
     """
 
-    def __init__(self, regex, flags=0, message=None):
+    def __init__(self, regex, flags=0, use_raw=False, message=None):
         if isinstance(regex, string_types):
             regex = re.compile(regex, flags)
         self.regex = regex
+        self.use_raw = use_raw
         self.message = message
 
     def __call__(self, form, field, message=None):
-        match = self.regex.match(field.data or "")
+        data = field.raw_data[0] if self.use_raw else field.data
```

Comment:
question to admins: Instead of validating only the first raw_data, maybe we need to test the regex against all items?